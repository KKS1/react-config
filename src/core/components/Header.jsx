import React, { Component } from "react";
import Helptip from "./Helptip";

const Header = props => {
  const { title, helptip } = props.schema;
  return (
    <div className="component r-header">
      <h1 className="question c-question__header js-substep-form-header" tabIndex="-1">
        <span className="mr-2">{title}</span>
        {helptip && <Helptip componentProps={helptip} />}
      </h1>
    </div>
  );
};

export default Header;
