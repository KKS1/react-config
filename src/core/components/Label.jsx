import React, { Component } from "react";

const Label = ({ schema, label, id }) => {
  return !schema && label ? <label htmlFor={id}>{label}</label> : "";
};

export default Label;
