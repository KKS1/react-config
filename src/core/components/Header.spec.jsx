import React, { Component } from "react";
import Label from "./Label";
import Header from "./Header";

describe("Header", () => {
  test("render_Header", () => {
    const props = {
      schema: {
        title: "loan question",
        helptip: [
          {
            type: "paragraph",
            text: "help tip paragraph content"
          }
        ]
      }
    };
    const wrapper = shallow(<Header {...props} />);
    const instance = wrapper.getElement();
    expect(instance).toMatchSnapshot();
  });
});
