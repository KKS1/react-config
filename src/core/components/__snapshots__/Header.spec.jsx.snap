// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render_Header 1`] = `
<div
  className="component r-header"
>
  <h1
    className="question c-question__header js-substep-form-header"
    tabIndex="-1"
  >
    <span
      className="mr-2"
    >
      loan question
    </span>
    <Helptip
      componentProps={
        Array [
          Object {
            "text": "help tip paragraph content",
            "type": "paragraph",
          },
        ]
      }
    />
  </h1>
</div>
`;
